napari/layers/utils/plane_manager.py

```diff
@@ -0,0 +1,75 @@
+from typing import Tuple
+
+import numpy as np
+from pydantic import validator
+
+from ...utils.events import EventedModel
+from ...utils.geometry import intersect_line_with_plane_3d
+from ...utils.misc import ensure_n_tuple
+
+
+class PlaneManager(EventedModel):
+    """Manages properties relating to plane rendering in 3D.
+
+    In this object, planes in 3D are defined by a position, a normal vector
+    and a thickness parameter.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A 3D position on the plane, defined in data coordinates.
+    normal : 3-tuple
+        A 3D unit vector normal to the plane, defined in data coordinates.
+    thickness : float
+        Thickness of the slice
+    enabled : bool
+        Whether plane rendering is enabled.
+    """
+
+    position: Tuple[float, float, float] = (0, 0, 0)
+    normal: Tuple[float, float, float] = (1, 0, 0)
+    thickness: float = 10.0
+    enabled: bool = False
+
+    @validator('position', 'normal', pre=True)
+    def _ensure_3_tuple(cls, v):
+        return ensure_n_tuple(v, n=3)
+
+    @validator('normal')
+    def _normalise_vector(cls, v):
+        return tuple(v / np.linalg.norm(v))
+
+    def shift_along_normal_vector(self, distance: float):
+        """Shift the plane along its normal vector by a given distance."""
+        self.position += distance * self.normal
+
+    def intersect_with_line(
+        self, line_position: np.ndarray, line_direction: np.ndarray
+    ) -> np.ndarray:
+        """Calculate a 3D line-plane intersection."""
+        return intersect_line_with_plane_3d(
+            line_position, line_direction, self.position, self.normal
+        )
+
+    @classmethod
+    def from_points(cls, points: np.ndarray):
```

Comment:
just an api thought:  I know we expect `np.ndarray` objects in many other places, but for very simple things like this, I wonder whether it's better to do `(cls, point0, point1, point2)`... it feels easier to use `from_points(*myarray)` (if you have an array already) than it is to have to do `from_points(np.ndarray([p1, p2, p3]))` (if you don't have a np.ndarray).  Plus, the shape of the expected object is a bit more explicit?